internal/generate/clusterserviceversion/bases/definitions/ast.go

```diff
@@ -18,15 +18,90 @@ import (
 	"errors"
 	"fmt"
 	"go/ast"
+	"strconv"
 	"strings"
 
+	"sigs.k8s.io/controller-tools/pkg/crd"
+	"sigs.k8s.io/controller-tools/pkg/loader"
 	"sigs.k8s.io/controller-tools/pkg/markers"
 )
 
-// getMarkedChildrenOfField collects all marked fields from type declarations starting at root in depth-first order.
-func (g generator) getMarkedChildrenOfField(root markers.FieldInfo) (map[string][]*fieldInfo, error) {
+// importIdents maps import identifiers to a list of corresponding path and file containing that path.
+// For example, consider the set of 2 files, one containing 'import foo "my/foo"' and the other 'import foo "your/foo"'.
+// Then the map would be: map["foo"][]struct{{f: (file 1), path: "my/foo"},{f: (file 2), path: "your/foo"}}.
+type importIdents map[string][]struct {
+	f    *ast.File
+	path string
+}
+
+// newImportIdents creates an importIdents from all imports in pkg.
+func newImportIdents(pkg *loader.Package) (importIdents, error) {
+	importIDs := make(map[string][]struct {
+		f    *ast.File
+		path string
+	})
+	for _, file := range pkg.Syntax {
+		for _, impSpec := range file.Imports {
+			val, err := strconv.Unquote(impSpec.Path.Value)
+			if err != nil {
+				return nil, err
+			}
+			// Most imports are not locally named, so the real package name should be used.
+			var impName string
+			if imp, hasImp := pkg.Imports()[val]; hasImp {
+				impName = imp.Name
+			}
+			// impSpec.Name will not be empty for locally named imports
+			if impSpec.Name != nil {
+				impName = impSpec.Name.Name
+			}
+			importIDs[impName] = append(importIDs[impName], struct {
+				f    *ast.File
+				path string
+			}{file, val})
+		}
+	}
+	return importIDs, nil
+}
+
+// findPackagePathForSelExpr returns the package path corresponding to the package name used in expr if it exists in im.
+func (im importIdents) findPackagePathForSelExpr(expr *ast.SelectorExpr) (pkgPath string) {
```

Comment:
Nit: Not a fan of this method name but I don't have a better option :) so not blocking this PR.